### PR TITLE
feature/observability

### PR DESCRIPTION
**Pull request by baywet**
related https://github.com/microsoft/kiota/issues/618
fixes https://github.com/microsoftgraph/msgraph-sdk-go-core/issues/31

### CODE REVIEW

**Comment by baywet** [comment]
TODO:
- uri decode the template in the method name
- get the object name for the request content and response

File: parameters_name_decoding_handler.go [hunk 1]
```diff
@@ -1,78 +1,96 @@
...
+
+// Intercept implements the RequestInterceptor interface and decodes the parameters name
+func (handler *ParametersNameDecodingHandler) Intercept(pipeline Pipeline, middlewareIndex int, req *nethttp.Request) (*nethttp.Response, error) {
+	reqOption, ok := req.Context().Value(parametersNameDecodingKeyValue).(parametersNameDecodingOptionsInt)
+	if !ok {
+		reqOption = &handler.options
+	}
+	obsOptions := GetObservabilityOptionsFromRequest(req)
+	ctx := req.Context()
+	var span trace.Span
```
Comment (rkodev) [inline parameters_name_decoding_handler.go:73]: minor: does not look like its necessary to declare 
`var span trace.Span`

File: parameters_name_decoding_handler.go [hunk 2]
```diff
@@ -1,78 +1,96 @@
...
+	if obsOptions != nil {
+		ctx, span = otel.GetTracerProvider().Tracer(obsOptions.GetTracerInstrumentationName()).Start(ctx, "ParametersNameDecodingHandler_Intercept")
```
Comment (baywet) [inline parameters_name_decoding_handler.go:75]:
```suggestion
		ctx, span := otel.GetTracerProvider().Tracer(obsOptions.GetTracerInstrumentationName()).Start(ctx, "ParametersNameDecodingHandler_Intercept")

```

File: parameters_name_decoding_handler.go (hunk 1 above)
Comment (baywet) [reply parameters_name_decoding_handler.go:73]:
```suggestion

```

File: parameters_name_decoding_handler.go [hunk 3]
```diff
@@ -1,78 +1,96 @@
...
+
+import (
+	nethttp "net/http"
+	"strconv"
+	"strings"
+
+	abs "github.com/microsoft/kiota-abstractions-go"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
```
Comment (baywet) [inline parameters_name_decoding_handler.go:11]:
```suggestion

```

**Review by rkodev — APPROVED** [review]

**Pull request merged**